### PR TITLE
Fix/apl 1032 fix import string columns

### PR DESCRIPTION
**Pull request by alzinchenko**
Fixed import string column;
Fixed process of cleaning up the previously imported data;

### CODE REVIEW

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/app/TrimServiceTest.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ void testInitWithExistingOldNotFinishedTrimEntry() {
         doReturn(7199).when(timeService).getEpochTime();
         mockTrimEntries(10000, 11000, 1000);
 
-        trimService.init(11999);
+        trimService.init(11999, 0);
```
Comment (AndrewBoyarsky) [inline apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/app/TrimServiceTest.java]: Please add test to validate new condition `if( lastTrimHeight < shardInitialBlockHeight{`

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import com.apollocurrency.aplwallet.apl.util.StringUtils;
+
+import javax.inject.Singleton;
+import java.util.Base64;
+
+@Singleton
+public class ValueParserImpl implements ValueParser {
+    @Override
+    public String parseStringObject(Object data) {
+        String stringObject = (String)data;
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java]: Check for NULL pls.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        }else{//string without quotes
+            stringValue = stringObject;
+        }
+        String value = stringValue.replaceAll("''", "'");
+        return value;
+    }
+
+    @Override
+    public Object[] parseArrayObject(Object data) {
+        String objectArray = (String)data;
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java]: NULL check would be safer for incoming param.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        } else {
+            actualArray = new Object[0];
+        }
+        return actualArray;
+    }
+
+    @Override
+    public byte[] parseBinaryObject(Object data) {
+        return Base64.getDecoder().decode(((String) data));
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java]: The same, let's throw exception explicitly for NULL value.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/CsvImporterImpl.java [hunk 5]
```diff
@@ -172,53 +174,27 @@ private long importCsv(String tableName, int batchLimit, boolean cleanTarget,
...
-                                }
-                            }
-                        } else {
-                            actualArray = new Object[0];
-                        }
+                        Object[] actualArray = parser.parseArrayObject(object);
                         SimpleResultSet.SimpleArray simpleArray = new SimpleResultSet.SimpleArray(actualArray);
                         preparedInsertStatement.setArray(i + 1, simpleArray);
                         row.put(columnName.toLowerCase(), actualArray);
+                    } else if (object != null && (meta.getColumnType(i + 1) == Types.VARCHAR || meta.getColumnType(i + 1) == Types.NVARCHAR)) {
```
Comment (silaev-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/CsvImporterImpl.java]: I suggest putting this condition in a separate method with an implicit name. For example:
isVarcharColumnTypeAndNonNull...

As a result, a further maintainer might understand it fast

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java (hunk 2 above)
Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java]: Please, consider this code:
final String stringObject = `Optional.ofNullable(data).map(String.class::cast).orElseThrow(() -> new IllegalArgumentException(""));`

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/shard/helper/csv/ValueParserTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+ */
+
+package com.apollocurrency.aplwallet.apl.core.shard.helper.csv;
+
+import com.apollocurrency.aplwallet.apl.core.shard.helper.ValueParserImpl;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigInteger;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (silaev-firstbridge) [inline apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/shard/helper/csv/ValueParserTest.java]: please, consider not using * in imports

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/shard/helper/csv/ValueParserTest.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        String o = "'APL-ZW95-E7B5-MVVP-CCBDT'";
+        String expected = o.substring(1, o.length()-1);
+        assertEquals(expected, parser.parseStringObject(o));
+    }
+
+    @Test
+    void parseStringObject() {
+        String o = "APL-ZW95-E7B5-MVVP-CCBDT";
+        assertEquals(o, parser.parseStringObject(o));
+    }
```
Comment (silaev-firstbridge) [inline apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/shard/helper/csv/ValueParserTest.java]: please, consider JUnit 5 Parameterized Tests to combine these 2 test cases into a single one

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/app/TrimServiceTest.java (hunk 1 above)
Comment (alzinchenko) [reply apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/app/TrimServiceTest.java]: done

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java (hunk 2 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java]: done

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java (hunk 4 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java]: done

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java (hunk 3 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/ValueParserImpl.java]: done

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/shard/helper/csv/ValueParserTest.java (hunk 7 above)
Comment (alzinchenko) [reply apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/shard/helper/csv/ValueParserTest.java]: done

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/CsvImporterImpl.java (hunk 5 above)
Comment (alzinchenko) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/CsvImporterImpl.java]: done

**Pull request merged**